src/Common/CustomLogStyle.php

```diff
@@ -0,0 +1,154 @@
+<?php
+namespace Robo\Common;
+
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\OutputStyle;
+
+/**
+ * Custom Log Styler. Styles may be overridden via constructor parameters.
+ */
+class CustomLogStyle implements LogStyleInterface
+{
+    const TASK_STYLE_INFO = 'fg=white;bg=cyan;options=bold';
+    const TASK_STYLE_SUCCESS = 'fg=white;bg=green;options=bold';
+    const TASK_STYLE_WARNING = 'fg=black;bg=yellow;options=bold;';
+    const TASK_STYLE_ERROR = 'fg=white;bg=red;options=bold';
+
+    protected $output;
+    protected $defaultStyles = [
+        '*' => 'info',
+    ];
+    protected $labelStyles = [
+        'error' => self::TASK_STYLE_ERROR,
+        'warning' => self::TASK_STYLE_WARNING,
+        'caution' => self::TASK_STYLE_WARNING,
+        'note' => self::TASK_STYLE_INFO,
+        'success' => self::TASK_STYLE_SUCCESS,
+    ];
+    protected $messageStyles = [
+        'error' => self::TASK_STYLE_ERROR,
+        'warning' => '',
+        'caution' => '',
+        'note' => '',
+        'success' => '',
+    ];
+
+    public function __construct(OutputInterface $output, $labelStyles = [], $messageStyles = [])
+    {
+        $this->output = $output;
+        $this->labelStyles = $labelStyles + $this->labelStyles;
+        $this->messageStyles = $messageStyles + $this->messageStyles;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function defaultStyles()
+    {
+        return $this->defaultStyles;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function style($context)
+    {
+        $context += ['_style' => []];
```

Comment:
We can't predict what variables folks will put in their context, so I added _ as a simple namespacing. We could rename this to something else -- `.style` or whatever. This is one of the classes that I hope to put into Symfony Console.  If this isn't part of Symfony, then of course we could just get rid of it completely and hardcode Robo-styled logging into the base class.